internal/services/nginx/nginx_deployment_resource.go

```diff
@@ -262,6 +285,24 @@ func (m DeploymentResource) Arguments() map[string]*pluginsdk.Schema {
 				}, false),
 		},
 
+		"web_application_firewall_settings": {
+			Type:     pluginsdk.TypeList,
+			Optional: true,
+			Elem: &pluginsdk.Resource{
+				Schema: map[string]*pluginsdk.Schema{
+					"activation_state": {
+						Type:     pluginsdk.TypeString,
+						Required: true,
+						ValidateFunc: validation.StringInSlice(
+							[]string{
+								"Enabled",
+								"Disabled",
+							}, false),
```

Comment:
this is a bool? so we should
```suggestion
						Type:     pluginsdk.TypeBool,
						Required: true,
```

and then in the provider code translate true/false to Enabled Disabled, unless there is going to be a 3rd state here?

Reply:
The state is an [enum](https://github.com/Azure/azure-rest-api-specs/blob/fa6c6015b1633a701edb629c1a759433aa1fdaf2/specification/nginx/resource-manager/NGINX.NGINXPLUS/preview/2024-09-01-preview/swagger.json#L2046).

Reply:
we initially designed the activation_state to be a bool in the schema but later changed to enum after recommendation from microsoft ARM team. Here is there reasoning 

> ARM recommends enums over booleans for future proof APIs. Standard guidance is: replace boolean/switch properties with a more meaningful enum whenever possible.
> 
> A boolean will forever have two valid values (true or false). A string enum type is always preferred. Also, properties should always provide better values just than True and False. For example two switches "isTypeA" and "isTypeB" should be replaced with one enum "type": [A, B, DefaultType]. Enums are always a more flexible and future proof option because they allow additional values to be added in the future in a non-breaking way, e.g. [Enabled, Disabled, Suspended, Deallocated].
> 
> Note: do NOT define a 'boolean enum' with two values 'True and False'. This might be easier to 'extend' in terms of types, but semantically its confusing, and no better than a boolean.
>

Reply:
As per our guidelines enums that are actually just booleans with different names should be booleans in schema with the correct name to telegraph what the boolean does https://github.com/hashicorp/terraform-provider-azurerm/blob/main/contributing/topics/reference-naming.md

Reply:
@katbyte updated the PR to use a bool instead of the enum